packages/cozy-intent/src/api/services/NativeService.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { WebviewRef } from 'api/models/environments'
+import { NativeEvent } from 'api/models/events'
+import { NativeMethodsRegister } from 'api/models/methods'
+import { NativeMessenger } from '../services/NativeMessenger'
+import { NativeService } from './NativeService'
+
+jest.mock('../services/NativeMessenger')
+
+const onMessageMock = jest.fn()
+class MockNativeMessenger extends NativeMessenger {
+  constructor(webviewRef: WebviewRef) {
+    super(webviewRef)
+  }
+
+  public postMessage = jest.fn()
+
+  public addMessageListener = jest.fn()
+
+  public onMessage = (event: NativeEvent): void => {
+    onMessageMock(event)
+  }
+}
```

Comment:
test too complicated, give me some time to study this 🙏

Reply:
I don't have a lot of time as I should close this PR today. Should we speak about this in a call?

Reply:
after study, I am ok to approve the PR.